4788/src/main/cpp/Shooter.cpp

```diff
@@ -1,82 +1,86 @@
+  setIndexVoltage = power;
+}
 
-  // double input = _shooterSystem._flyWheel.encoder->GetAngularVelocity();
+void Shooter::setPID(double goal, double dt) {
+  _state = ShooterState::kPID;
+  angularVelocityGoal = goal;
+}
 
-  // ControlMap::error = ControlMap::goal - input;
-  // ControlMap::derror = (ControlMap::error - ControlMap::previousError) / dt;
-  // ControlMap::sum = ControlMap::sum + ControlMap::error * dt;
+void Shooter::updateShooter(double dt) {
+  double pidSetPower = calculatePID(angularVelocityGoal, dt);
 
-  // ControlMap::ouput = ControlMap::kp * ControlMap::error + ControlMap::ki * ControlMap::sum + ControlMap::kd * ControlMap::derror;
-  // ControlMap::previousError = ControlMap::error;
 
-  // return ControlMap::output;
-}
+  switch (_state) {
+  case ShooterState::kManual:
+    _shooterSystem.shooterGearbox.transmission->SetVoltage(setFlyWheelVoltage * 12);
+
+    _shooterSystem.indexWheel.Set(setIndexVoltage);
+
+    break;
+
+  case ShooterState::kIdle:
 
+    _shooterSystem.indexWheel.Set(setIndexVoltage);
+    _shooterSystem.shooterGearbox.transmission->SetVoltage(0);
 
-/**
-  * Left trigger controls the shooter manually
-  */
-void Shooter::manualControl(double dt) {
-  shooterManualSpeed = fabs(_contGroup.Get(ControlMap::shooterManualSpin)) > ControlMap::triggerDeadzone ? _contGroup.Get(ControlMap::shooterManualSpin) : 0;
+    break;
 
-  _shooterSystem.shooterGearbox.transmission->SetVoltage(shooterManualSpeed);
+  case ShooterState::kPID:
+    _shooterSystem.indexWheel.Set(setIndexVoltage);
+
+    _shooterSystem.shooterGearbox.transmission->SetVoltage(pidSetPower * 12);
+    std::cout << "shooter speed" << _shooterSystem.rightFlyWheelMotor.GetEncoderAngularVelocity() << std::endl; 
+    break;
+
+  default:
+    std::cout << "in default case, somthing is wrong" << std::endl;
+    break;
+  }
 }
 
-/**
-  * for testing the shooter
-  */
-void Shooter::testing(double dt) {
+void Shooter::update(double dt) {
+  updateShooter(dt);
+}
 
-  shooterTestingSpeed = fabs(_contGroup.Get(ControlMap::shooterManualSpin)) > ControlMap::triggerDeadzone ? _contGroup.Get(ControlMap::shooterManualSpin) : 0;
 
-  // _shooterSystem.shooterGearbox.transmission->SetVoltage(shooterTestingSpeed);
+double sum = 0;
+double previousError = 0;
+double Shooter::calculatePID(double angularVelocity, double dt) {
+  std::cout << "goal: " << angularVelocityGoal << std::endl;
+  double input = (_shooterSystem.leftFlyWheelMotor.GetEncoderAngularVelocity());
+  std::cout << "angular velocity" << input << std::endl;
+  double error = angularVelocityGoal - input;
+  double derror = (error - previousError) / dt;
+  sum += error * dt;
 
-  _shooterSystem.leftFlyWheelMotor.Set(shooterTestingSpeed);
-  _shooterSystem.rightFlyWheelMotor.Set(shooterTestingSpeed);
-  _shooterSystem.centerFlyWheelMotor.Set(shooterTestingSpeed);
+  double output = ControlMap::kp * error + ControlMap::ki * sum + (ControlMap::kd) * derror;
```

Comment:
If you store your PID values in controlmap, then you need to be more descriptive than just kP kI kD.
E.g, if someone accidentally used these values for drivetrain PID, wouldn't be that fun.

Possibly store in a struct